src/commcare_cloud/ansible/plugins/lookup/cchq_aws_secret.py

```diff
@@ -0,0 +1,91 @@
+import json
+import logging
+import os
+import cPickle
+import time
+
+from ansible.plugins.lookup import aws_secret
+from cryptography.fernet import Fernet, InvalidToken
+
+__metaclass__ = type
+
+
+DOCUMENTATION = aws_secret.DOCUMENTATION
+RETURN = aws_secret.RETURN
+
+LOOKUP_IN_PROGRESS_ON_ANOTHER_FORK = b'FETCHING'
+
+
+class LookupModule(aws_secret.LookupModule):
+    """
+    commcare-cloud's wrapper around aws_secret
+
+    Adds commcare-cloud specific caching to aws_secret lookup plugin for big speedup
+    Also assumes the value is json and returns the string json-decoded unlike aws_secret.
+
+    To cache even between forks of the same process/run, we use a one-time process-wide key
+    to encrypt values and write them to disk in files <env_dir>/.generated/.
+    After the run is over, the encryption key is lost to time and the files are useless
+    and indecipherable even by the person who ran it.
+    """
+    def run(self, terms, variables, **kwargs):
+        terms = tuple(terms)
+        term, = terms
+        value = self.get_cache(term, inventory_dir=variables['inventory_dir'])
+        try:
+            if value is Ellipsis:
+                logging.debug('Fetching secret {}'.format(terms))
+                try:
+                    value = super(LookupModule, self).run(terms, variables, **kwargs)
+                except Exception as e:
+                    logging.debug('Caching error fetching secret: {}'.format(e))
+                    value = e
+                self.set_cache(term, value, inventory_dir=variables['inventory_dir'])
```

Comment:
It's not clear what `term` and `value` are. Is there one term per secret, or something else? If one per secret, does that mean there is a separate cache file per secret?